python/ray/experimental/workflow/storage/base.py

```diff
@@ -0,0 +1,178 @@
+import abc
+from abc import abstractmethod
+
+from dataclasses import dataclass
+
+import ray
+from ray.experimental.workflow.common import StepID
+from typing import Any, Callable, List, Dict, Tuple
+
+ArgsType = Tuple[List[Any], Dict[str, Any]]  # args and kwargs
+
+
+@dataclass
+class StepStatus:
+    # does the step output checkpoint exists?
+    output_object_exists: bool
+    # does the step output metadata exists?
+    output_metadata_exists: bool
+    # does the step input metadata exists?
+    input_metadata_exists: bool
+    # does the step input argument exists?
+    args_exists: bool
+    # does the step function body exists?
+    func_body_exists: bool
+
+
+class Storage(metaclass=abc.ABCMeta):
+    @abstractmethod
+    def load_step_input_metadata(self, workflow_id: str,
+                                 step_id: StepID) -> Dict[str, Any]:
+        """Dump the input metadata of a step.
+
+        Args:
+            workflow_id: ID of the workflow job.
+            step_id: ID of the step.
+
+        Returns:
+            A metadata dict.
```

Comment:
What happens if the step is not found?

Reply:
It raises an exception. I'll unify the exception type.